File: backend/CuraSphere.Domain/Entities/ClinicalServices.cs
using System;

namespace CuraSphere.Domain.Entities;

public class PharmacyStock : BaseEntity
{
    public long DrugId { get; set; }
    public string DrugCode { get; set; } = string.Empty;
    public string DrugName { get; set; } = string.Empty;
    public string GenericName { get; set; } = string.Empty;
    public string DosageForm { get; set; } = "Tablet"; // Tablet, Syrup, Injection, Ointment
    public int QuantityInStock { get; set; }
    public int MinimumThreshold { get; set; } = 50;
    public decimal UnitPrice { get; set; }
    public DateTime ExpiryDate { get; set; }
    public string Status { get; set; } = "Available"; // Available, Low Stock, Expired
}

public class SurgicalProcedure : BaseEntity
{
    public long SurgeryId { get; set; }
    public string SurgeryNumber { get; set; } = string.Empty;
    public long PatientId { get; set; }
    public virtual Patient Patient { get; set; } = null!;
    
    public string ProcedureName { get; set; } = string.Empty;
    public string OtRoomNumber { get; set; } = "OT-1";
    public long LeadSurgeonId { get; set; }
    public virtual Doctor LeadSurgeon { get; set; } = null!;
    
    public string AnesthetistName { get; set; } = string.Empty;
    public DateTime ScheduledStartTime { get; set; }
    public DateTime? ActualEndTime { get; set; }
    public string Status { get; set; } = "Scheduled"; // Scheduled, In Surgery, Completed, Cancelled
}

public class IcuPatientVital : BaseEntity
{
    public long VitalId { get; set; }
    public long PatientId { get; set; }
    public virtual Patient Patient { get; set; } = null!;
    
    public string BedNumber { get; set; } = "ICU-01";
    public string BloodPressure { get; set; } = "120/80";
    public int HeartRate { get; set; } = 75;
    public int SpO2 { get; set; } = 98; // Oxygen Saturation %
    public int GcsScore { get; set; } = 15; // Glasgow Coma Scale
    public decimal Temperature { get; set; } = 98.6m;
    public DateTime RecordedAt { get; set; } = DateTime.UtcNow;
    public string AlertStatus { get; set; } = "Normal"; // Normal, Warning, Critical
}

File: frontend/src/services/clinicalServices.ts
import api from "../lib/api";

export interface PharmacyStockItem {
  drugId: number;
  drugCode: string;
  drugName: string;
  genericName: string;
  dosageForm: string;
  quantityInStock: number;
  minimumThreshold: number;
  unitPrice: number;
  expiryDate: string;
  status: string;
}

export interface SurgicalScheduleItem {
  surgeryId: number;
  surgeryNumber: string;
  patientId: number;
  patientName: string;
  medicalRecordNumber: string;
  procedureName: string;
  otRoomNumber: string;
  leadSurgeonName: string;
  anesthetistName: string;
  scheduledStartTime: string;
  status: string;
}

export interface IcuPatientVitalItem {
  vitalId: number;
  patientId: number;
  patientName: string;
  medicalRecordNumber: string;
  bedNumber: string;
  bloodPressure: string;
  heartRate: number;
  spO2: number;
  gcsScore: number;
  temperature: number;
  recordedAt: string;
  alertStatus: string;
}

export const clinicalServices = {
  getPharmacyStock: async (search?: string): Promise<PharmacyStockItem[]> => {
    const params: Record<string, string> = {};
    if (search) params.search = search;

    const res = await api.get("/clinicalservices/pharmacy/stock", { params });
    return res.data?.data ?? [];
  },

  getSurgerySchedules: async (search?: string): Promise<SurgicalScheduleItem[]> => {
    const params: Record<string, string> = {};
    if (search) params.search = search;

    const res = await api.get("/clinicalservices/surgery/schedules", { params });
    return res.data?.data ?? [];
  },

  getIcuVitals: async (search?: string): Promise<IcuPatientVitalItem[]> => {
    const params: Record<string, string> = {};
    if (search) params.search = search;

    const res = await api.get("/clinicalservices/icu/vitals", { params });
    return res.data?.data ?? [];
  },
};
